dask/async.py

```diff
@@ -260,23 +262,24 @@ def execute_task(key, task, data, queue, get_id, raise_on_exception=False):
     _execute_task - actually execute task
     """
     try:
+        task, data = unmarshall(task_info)
         result = _execute_task(task, data)
         id = get_id()
-        result = key, result, None, id
-    except Exception as e:
-        if raise_on_exception:
-            raise
-        exc_type, exc_value, exc_traceback = sys.exc_info()
-        tb = ''.join(traceback.format_tb(exc_traceback))
-        result = key, e, tb, None
-    try:
-        queue.put(result)
+        result = marshall((result, None, id))
     except Exception as e:
         if raise_on_exception:
             raise
         exc_type, exc_value, exc_traceback = sys.exc_info()
         tb = ''.join(traceback.format_tb(exc_traceback))
-        queue.put((key, e, tb, None))
+        try:
+            result = marshall((e, tb, None))
+        except Exception as e:
```

Comment:
Is there a chance that serializing the exception and traceback might fail? If not, I'd opt to remove this block.